core/src/Threads/Kokkos_ThreadsTeam.hpp

```diff
@@ -815,7 +815,7 @@ class TeamPolicyInternal<Kokkos::Threads, Properties...>
     }
 
     int new_chunk_size = 1;
-    while (new_chunk_size * 100 * concurrency < m_league_size)
+    while ((int64_t)new_chunk_size * 100 * concurrency < (int64_t)m_league_size)
```

Comment:
Hmmm... I would at least also cast the comparison in line 822. I would prefer just declaring concurrency as `int64_t`.

Reply:
You prefer this?
```diff
diff --git a/core/src/Threads/Kokkos_ThreadsTeam.hpp b/core/src/Threads/Kokkos_ThreadsTeam.hpp
index 1e4b2dad..5b0cb089 100644
--- a/core/src/Threads/Kokkos_ThreadsTeam.hpp
+++ b/core/src/Threads/Kokkos_ThreadsTeam.hpp
@@ -806,7 +806,7 @@ class TeamPolicyInternal<Kokkos::Threads, Properties...>
  private:
   /** \brief finalize chunk_size if it was set to AUTO*/
   inline void set_auto_chunk_size() {
-    int concurrency = traits::execution_space::concurrency() / m_team_alloc;
+    int64_t concurrency = traits::execution_space::concurrency() / m_team_alloc;
     if (concurrency == 0) concurrency = 1;

     if (m_chunk_size > 0) {
@@ -815,11 +815,11 @@ class TeamPolicyInternal<Kokkos::Threads, Properties...>
     }

     int new_chunk_size = 1;
-    while ((int64_t)new_chunk_size * 100 * concurrency < (int64_t)m_league_size)
+    while (new_chunk_size * 100 * concurrency < (int64_t)m_league_size)
       new_chunk_size *= 2;
     if (new_chunk_size < 128) {
       new_chunk_size = 1;
-      while ((new_chunk_size * 40 * concurrency < m_league_size) &&
+      while ((new_chunk_size * 40 * concurrency < (int64_t)m_league_size) &&
              (new_chunk_size < 128))
         new_chunk_size *= 2;
     }

Reply:
Yes, if casting the right-hand-side is still necessary. I was hoping that we could avoid that.

Reply:
Updated